packages/devtools_app/test/shared/editable_list_test.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../test_utils/test_utils.dart';
+
+void main() {
+  const windowSize = Size(1000.0, 1000.0);
+  const label = 'This is the EditableList label';
+  ValueListenable<List<String>> entries([
+    List<String> values = const <String>[],
+  ]) {
+    return ListValueNotifier<String>(values);
+  }
+
+  setGlobal(IdeTheme, getIdeTheme());
+
+  group('EditableList', () {
+    testWidgetsWithWindowSize('shows the label', windowSize,
+        (WidgetTester tester) async {
+      await tester.pumpWidget(
+        wrap(
+          EditableList(entries: entries([]), textFieldLabel: label),
+        ),
+      );
+      expect(find.text(label), findsOneWidget);
+    });
+
+    testWidgetsWithWindowSize('triggers add', windowSize,
+        (WidgetTester tester) async {
+      const valueToAdd = 'this is the value that will be added';
+      String? entryThatWasRequestedForAdd;
+      final widget = EditableList(
+        entries: entries(),
+        textFieldLabel: label,
+        onEntryAdded: (e) {
+          entryThatWasRequestedForAdd = e;
+        },
+      );
+      await tester.pumpWidget(
+        wrap(widget),
+      );
+
+      final actionBar = find.byType(EditableListActionBar);
+      final textField = find.descendant(
+        of: actionBar,
+        matching: find.byType(TextField),
+      );
+      final addEntryButton = find.descendant(
+        of: actionBar,
+        matching: find.text('Add'),
+      );
+      await tester.enterText(textField, valueToAdd);
+      await tester.tap(addEntryButton);
+      await tester.pump();
+
+      expect(entryThatWasRequestedForAdd, equals(valueToAdd));
+    });
+
+    testWidgetsWithWindowSize('triggers refresh', windowSize,
+        (WidgetTester tester) async {
+      var refreshWasCalled = false;
+      final widget = EditableList(
+        entries: entries(),
+        textFieldLabel: label,
+        onRefresh: () {
+          refreshWasCalled = true;
+        },
+      );
+      await tester.pumpWidget(
+        wrap(widget),
+      );
+      final refreshButton = find.byType(RefreshButton);
+
+      await tester.tap(refreshButton);
+      await tester.pump();
+
+      expect(refreshWasCalled, isTrue);
+    });
+
+    testWidgetsWithWindowSize('triggers remove', windowSize,
+        (WidgetTester tester) async {
+      const valueToRemove = 'this is the value that will be removed';
+      String? entryThatWasRemoved;
+      final widget = EditableList(
+        entries: entries([valueToRemove]),
+        textFieldLabel: label,
+        onEntryRemoved: (e) {
+          entryThatWasRemoved = e;
+        },
+      );
+      await tester.pumpWidget(
+        wrap(widget),
+      );
+      final row = find.byType(EditableListRow);
+      final removeButton = find.descendant(
+        of: row,
+        matching: find.byType(EditableListRemoveDirectoryButton),
+      );
```

Comment:
you can also just find.byType(EditableListRemoveDirectoryButton) directly instead of finding the row first and then grabbing the descendant